### PR TITLE
Changed so can pass arguments for different environments

### PR DESCRIPTION
**Pull request by MIslam92**
# Motivation and Context
Need to configure for different environments so changed to an argument based script

# What has changed
Passing in arguments now i.e. url, username and password

# How to test?
- Simply run `pipenv run python ./scripts/config_comms_templates.py http://localhost:8182 admin secret` to get templates into your db first.
- Then run`pipenv run python ./scripts/config_existing_comms_template.py http://localhost:8182 admin secret` to add the updated templates
- Check all templates are in your db based on JSON schemas that get loaded in.

# Links
Card: https://trello.com/c/pUPbT4Q0/336-ensure-all-comms-templates-are-configured-in-production-m

### CODE REVIEW

File: scripts/config_comms_templates.py [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
 # Run with the cmd pipenv run python scripts/config_comms_templates.py
 with open('scripts/templates_to_load.json') as read_file:
     contents = json.load(read_file)
+    username = args.user
+    password = args.password
 
-    region = "http://localhost:8182/classificationtypes/REGION"
-
-    legal_basis = "http://localhost:8182/classificationtypes/LEGAL_BASIS"
+    region = f'{args.default_url}/classificationtypes/REGION'
```
Comment (andrewmil) [inline scripts/config_comms_templates.py]: For consistency I'd define `args.default_url` as a variable

File: scripts/config_comms_templates.py [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+parser = argparse.ArgumentParser(description='Pass in a service URL')
+default_url = parser.add_argument('default_url', help='Main URL')
+arg_user = parser.add_argument('user', help='User')
+arg_password = parser.add_argument('password', help='Pass')
+args = parser.parse_args()
+
 # Script which adds new templates to the comms-service db along with the associated classifiers
 # Run with the cmd pipenv run python scripts/config_comms_templates.py
 with open('scripts/templates_to_load.json') as read_file:
     contents = json.load(read_file)
```
Comment (andrewmil) [inline scripts/config_comms_templates.py]: You only need this line inside the `with` context manager. The rest doesn't need to be indented

File: scripts/config_comms_templates.py [hunk 3]
```diff
@@ -1,28 +1,34 @@
+import argparse
 import requests
-from flask import json, current_app
+from flask import json
```
Comment (andrewmil) [inline scripts/config_comms_templates.py]: I'd just use the standard library `json` here

**Review by andrewmil — CHANGES_REQUESTED** [review]
You'll probably want a separate pipfile in the scripts folder to include your dependencies in.
`requests` isn't defined in the top level one

**Review by tej99 — APPROVED** [review]

File: scripts/config_comms_templates.py [hunk 4]
```diff
@@ -1,28 +1,36 @@
...
-    requests.post(communication, auth=(user, pwd), headers=headers)
+requests.post(region, auth=(username, password), headers=headers)
+requests.post(legal_basis, auth=(username, password), headers=headers)
+requests.post(communication, auth=(username, password), headers=headers)
 
-    for content in contents:
-        url = "http://localhost:8182/templates"
-        r = requests.post(url, json=content, auth=(user, pwd), headers=headers)
+for content in contents:
+    url = f'{arg_url}/templates'
```
Comment (andrewmil) [inline scripts/config_comms_templates.py]: This can be defined before the `for` loop

File: scripts/config_existing_comms_template.py [hunk 5]
```diff
@@ -1,19 +1,27 @@
+import argparse
 import requests
-from flask import json, current_app
+import json
 
 headers = {'Content-type': 'application/json'}
 
+parser = argparse.ArgumentParser(description='Pass in a service URL')
```
Comment (andrewmil) [inline scripts/config_existing_comms_template.py]: Description is out of date as we're now also passing in user/password

File: scripts/config_comms_templates.py [hunk 6]
```diff
@@ -1,28 +1,36 @@
...
 
+parser = argparse.ArgumentParser(description='Pass in a service URL')
+default_url = parser.add_argument('default_url', help='Main URL')
+arg_user = parser.add_argument('user', help='User')
+arg_password = parser.add_argument('password', help='Pass')
+args = parser.parse_args()
+
 # Script which adds new templates to the comms-service db along with the associated classifiers
-# Run with the cmd pipenv run python scripts/config_comms_templates.py
+# Run with the cmd pipenv run python scripts/config_comms_templates.py, 'url', 'user', 'password'
```
Comment (andrewmil) [inline scripts/config_comms_templates.py]: Don't need the commas here

File: scripts/config_existing_comms_template.py [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
+default_url = parser.add_argument('default_url', help='Main URL')
+arg_user = parser.add_argument('user', help='User')
+arg_password = parser.add_argument('password', help='Pass')
+args = parser.parse_args()
+
 # Script which updates existing templates in the comms-service db
-# Run with the cmd pipenv run python scripts/config_existing_comms_templates.py
+# Run with the cmd pipenv run python ./scripts/config_existing_comms_templates.py, 'url', 'user', 'password'
```
Comment (andrewmil) [inline scripts/config_existing_comms_template.py]: no need for commas here

File: scripts/config_comms_templates.py [hunk 8]
```diff
@@ -1,28 +1,36 @@
+import argparse
 import requests
-from flask import json, current_app
+import json
 
 headers = {'Content-type': 'application/json'}
 
+parser = argparse.ArgumentParser(description='Pass in a service URL')
```
Comment (andrewmil) [inline scripts/config_comms_templates.py]: Description is out of date as we're now also passing in user/password

**Review by andrewmil — APPROVED** [review]
Tested and works fine, left a few small comments but nothing serious

**Review by shahi645 — APPROVED** [review]
Approving to merge 🌚

**Pull request merged**